Atomic/syscontainers.py

```diff
@@ -923,7 +923,12 @@ def _get_containers_at(self, checkouts, are_preinstalled, containers=None):
             if fullpath.endswith(".0") or fullpath.endswith(".1"):
                 continue
 
-            with open(os.path.join(fullpath, "info"), "r") as info_file:
+            try:
+                fd = open(os.path.join(fullpath, "info"), "r")
+            except FileNotFoundError:
+                # Something got wrong?
+                continue
+            with fd as info_file:
```

Comment:
I did it this way because exceptions are more pythonic. I don't mind rewriting the code if you prefer in your codebase using `if`s over exceptions.

The descriptor would not leak: if the file does not exist, the descriptor would not be created; if the file exists, the descriptor will be closed within with statement:

```python
In [10]: fd = open("/etc/passwd", "r")

In [11]: with fd as x:
   ....:     x.read(3)
   ....:

In [12]: x
Out[12]: <closed file '/etc/passwd', mode 'r' at 0x7f76252cf420>

In [13]: fd
Out[13]: <closed file '/etc/passwd', mode 'r' at 0x7f76252cf420>
```